src/modules/mahjong/Hand/hand.ts

```diff
@@ -0,0 +1,107 @@
+import Tile from '../Tile/tile';
+import Wall from '../Wall/wall';
+import DeadPile from '../DeadPile/deadPile';
+import SimpleTileTypes from '../Tile/types/simpleTileTypes';
+import HonorTileTypes from '../Tile/types/honorTileTypes';
+import BonusTileTypes from '../Tile/types/bonusTileTypes';
+
+interface SortHandWeights {
+  [SimpleTileTypes.DOT]: number;
+  [SimpleTileTypes.BAMBOO]: number;
+  [SimpleTileTypes.CHARACTER]: number;
+  [HonorTileTypes.EAST]: number;
+  [HonorTileTypes.SOUTH]: number;
+  [HonorTileTypes.WEST]: number;
+  [HonorTileTypes.NORTH]: number;
+  [HonorTileTypes.GREEN_DRAGON]: number;
+  [HonorTileTypes.RED_DRAGON]: number;
+  [HonorTileTypes.WHITE_DRAGON]: number;
+  [BonusTileTypes.FLOWER]: number;
+  [BonusTileTypes.SEASON]: number;
+}
+
+class Hand {
+  private hand: Tile[];
+
+  private wall: Wall;
+
+  private deadPile: DeadPile;
+
```

Comment:
I think the hand probably won't have a reference to wall and deadpile as it might be too coupled. Perhaps pass in reference to the methods as arguments that use wall/deadpile